examples/statemachine/src/language-server/statemachine-validator.ts

```diff
@@ -54,4 +67,72 @@ export class StatemachineValidator {
             }
         }
     }
+
+    /**
+     * Checks if there are unreached state.
+     * @param statemachine the statemachine to check
+     * @param accept the acceptor to report errors
+     */
+    checkUnreachedStates(statemachine: Statemachine, accept: ValidationAcceptor): void {
```

Comment:
Sorry I'm not following, what's the initial state in the above code block? If it's `Next`, then it should be reached, shouldn't it?

Reply:
See the following image:

![image](https://github.com/eclipse-langium/langium/assets/4377073/25a0de15-8ba9-4261-934e-f691a518c1a5)

`Initial` should not be marked as unreachable, as it's the initial state - it's always reachable. Meanwhile, `Next`, which is actually unreachable in this statemachine, isn't marked as being such.

Note that I would recommend to just use `'hint'` instead of `'warning'` as the severity here.

Reply:
It should be resolve in 86306a2.